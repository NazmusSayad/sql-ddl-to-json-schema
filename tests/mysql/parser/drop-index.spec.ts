import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should drop index.': {
    queries: ['drop index i_oid on people;'],
  },

  'Parser: Should drop index with algorithm option.': {
    queries: [
      'drop index i_oid on people algorithm default lock none;',
      'drop index i_oid on people algorithm=default lock=none;',
      'drop index i_oid on people algorithm = default lock = none;',
    ],
  },

  'Parser: Should drop index with lock option': {
    queries: [
      'drop index i_oid on people lock default;',
      'drop index i_oid on people lock=default;',
      'drop index i_oid on people lock = default;',
    ],
  },
});
