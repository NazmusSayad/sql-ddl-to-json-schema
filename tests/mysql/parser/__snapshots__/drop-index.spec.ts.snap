// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should drop index with algorithm option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_INDEX",
        "def": {
          "index": "i_oid",
          "table": "people",
          "options": [
            {
              "id": "P_INDEX_ALGORITHM_OPTION",
              "def": {
                "algorithm": "default"
              }
            },
            {
              "id": "P_LOCK_OPTION",
              "def": {
                "lock": "none"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop index with algorithm option. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_INDEX",
        "def": {
          "index": "i_oid",
          "table": "people",
          "options": [
            {
              "id": "P_INDEX_ALGORITHM_OPTION",
              "def": {
                "algorithm": "default"
              }
            },
            {
              "id": "P_LOCK_OPTION",
              "def": {
                "lock": "none"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop index with algorithm option. 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_INDEX",
        "def": {
          "index": "i_oid",
          "table": "people",
          "options": [
            {
              "id": "P_INDEX_ALGORITHM_OPTION",
              "def": {
                "algorithm": "default"
              }
            },
            {
              "id": "P_LOCK_OPTION",
              "def": {
                "lock": "none"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop index with lock option 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_INDEX",
        "def": {
          "index": "i_oid",
          "table": "people",
          "options": [
            {
              "id": "P_LOCK_OPTION",
              "def": {
                "lock": "default"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop index with lock option 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_INDEX",
        "def": {
          "index": "i_oid",
          "table": "people",
          "options": [
            {
              "id": "P_LOCK_OPTION",
              "def": {
                "lock": "default"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop index with lock option 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_INDEX",
        "def": {
          "index": "i_oid",
          "table": "people",
          "options": [
            {
              "id": "P_LOCK_OPTION",
              "def": {
                "lock": "default"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop index. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_INDEX",
        "def": {
          "index": "i_oid",
          "table": "people",
          "options": []
        }
      }
    }
  ]
}
`;
